src/webui/api/torrentscontroller.cpp

```diff
@@ -252,6 +262,103 @@ namespace
             idList << BitTorrent::TorrentID::fromString(hash);
         return idList;
     }
+
+    QJsonObject serializePartialTorrentInfo(const BitTorrent::InfoHash &infoHash)
+    {
+        return QJsonObject {
+            {KEY_TORRENT_INFOHASHV1, infoHash.v1().toString()},
+            {KEY_TORRENT_INFOHASHV2, infoHash.v2().toString()},
+            {KEY_TORRENT_ID, infoHash.toTorrentID().toString()},
+        };
+    }
+
+    QJsonObject serializeTorrentInfo(const BitTorrent::TorrentInfo &info)
+    {
+        QJsonArray files;
+        for (int fileIndex = 0; fileIndex < info.filesCount(); ++fileIndex)
+        {
+            files << QJsonObject
+            {
+                {KEY_FILE_INDEX, fileIndex},
+                {KEY_FILE_NAME, info.filePath(fileIndex).toString()},
```

Comment:
Yes, I think that would be the expected behavior. What do you think?

Reply:
If paths were provided in a unified way, it would make it easier for client apps to handle them.

Reply:
The `torrents/properties` and `app/preferences` APIs currently send back a save path. From reading the code it's not clear to me if those paths are OS specific or not, but my goal would be to follow the same behavior. I don't have a windows OS to test on.

Reply:
> The `torrents/properties` and `app/preferences` APIs currently send back a save path. From reading the code it's not clear to me if those paths are OS specific or not, but my goal would be to follow the same behavior.

They send paths in OS specific form for some (IMO, invalid) reason.
But `torrents/files` sends unified paths and I would follow it:
https://github.com/qbittorrent/qBittorrent/blob/1c43286616e5c1a6420f9d1049ae860dbe0c2d6f/src/webui/api/torrentscontroller.cpp#L719-L720

Reply:
Hmm, but doesn't it make sense to provide paths in a Windows-specific format on Windows?

Reply:
This makes sense when displaying paths to the user. But API is for using by applications. I still believe that for the convenience of processing by applications, it is much better if the paths are transmitted in a unified form. Otherwise, any path processing related code should take care about the fact that the path may be presented in different form depending on the server OS.

Reply:
I can get behind that logic. Will push up a change.